src/west/commands/project.py

```diff
@@ -726,6 +727,19 @@ def do_run(self, args, user_args):
 # Private helper routines.
 #
 
+def _clean_west_refspace(project):
+    west_refs_cmd = ('for-each-ref --format="%(refname)" -- ' +
```

Comment:
Short (py)doc explaining the rationale. AFAIK:
- garbage collect and save disk space after (configurable!) reflog expiration
- avoid potentially large and unbounded number of refs

Reply:
> Short (py)doc explaining the rationale. AFAIK:
> 
>     * garbage collect and save disk space after (configurable!) reflog expiration
> 
>     * avoid potentially large and unbounded number of refs

I don't think our rationale is to save disk space.
The main purposes is to:
1) Be able to fetch a SHA from a git server (reason for `refs/west`)
2) minimize when fetching is needed by keeping a reference to SHAs (west smart fetching)
  Also achieved `refs/west`
3) Avoid `refs/west` shows up, as example in `git log`

So the cleanup is mostly for 3., but the reflog part is to hopefully keep the refs alive for longer time. (not cleaning `refs/west` would have kept them alive until user did a manual clean)

But will add some description :wink: